runtime/src/iree/hal/utils/libmpi_test.cc

```diff
@@ -13,40 +13,97 @@
 
 namespace {
 
-TEST(libmpi, DynamicLoadLibraryAndSymbols) {
-  iree_dynamic_library_t* library = NULL;
-  iree_hal_mpi_dynamic_symbols_t symbols = {0};
-  iree_status_t status =
-      iree_hal_mpi_library_load(iree_allocator_system(), &library, &symbols);
-  if (!iree_status_is_ok(status)) {
-    iree_status_fprint(stderr, status);
-    iree_status_ignore(status);
-    std::cerr << "Symbols cannot be loaded, skipping test.";
-    GTEST_SKIP();
+const int MPI_SUCCESS = 0;
+
+class LibmpiTest : public ::testing::Test {
+ protected:
+  static void SetUpTestSuite() {
+    iree_status_t status =
+        iree_hal_mpi_library_load(iree_allocator_system(), &library, &symbols);
+
+    if (!iree_status_is_ok(status)) {
+      iree_status_fprint(stderr, status);
+      iree_status_ignore(status);
+    } else {
+      EXPECT_EQ(symbols.MPI_Init(NULL, NULL), MPI_SUCCESS);
+    }
   }
 
-  IREE_EXPECT_OK(
-      MPI_RESULT_TO_STATUS(&symbols, MPI_Init(NULL, NULL), "MPI_Init"));
+  void SetUp() override {
+    if (!library) GTEST_SKIP() << "No MPI library available. Skipping suite.";
+    bool has_symbols = false;
+    unsigned char *syms_ptr = (unsigned char *)&symbols;
+    for (int i = 0; i < sizeof(symbols); i++) {
+      if (*(syms_ptr + i) != 0) {
+        has_symbols = true;
+        break;
+      }
+    }
+    if (!has_symbols)
+      GTEST_SKIP() << "MPI library failed to load symbols. Skipping suite";
 
-  int size = 0;
-  IREE_EXPECT_OK(MPI_RESULT_TO_STATUS(
-      &symbols, MPI_Comm_size(IREE_MPI_COMM_WORLD(&symbols), &size),
-      "MPI_Comm_size"));
+    IREE_EXPECT_OK(MPI_RESULT_TO_STATUS(
+        &symbols, MPI_Comm_size(IREE_MPI_COMM_WORLD(&symbols), &world_size),
+        "MPI_Comm_size"));
 
-  int rank = 0;
-  IREE_EXPECT_OK(MPI_RESULT_TO_STATUS(
-      &symbols, MPI_Comm_rank(IREE_MPI_COMM_WORLD(&symbols), &rank),
-      "MPI_Comm_rank"));
+    IREE_EXPECT_OK(MPI_RESULT_TO_STATUS(
+        &symbols, MPI_Comm_rank(IREE_MPI_COMM_WORLD(&symbols), &rank),
+        "MPI_Comm_rank"));
+
+    EXPECT_LT(rank, world_size);
+  }
 
-  EXPECT_LT(rank, size);
+  void TearDown() override {}
 
+  static void TearDownTestSuite() {
+    if (!library) return;
+
+    IREE_EXPECT_OK(
+        MPI_RESULT_TO_STATUS(&symbols, MPI_Finalize(), "MPI_Finalize"));
+
+    memset(&symbols, 0, sizeof(symbols));
+    if (library) iree_dynamic_library_release(library);
+  }
+
+ protected:
+  static iree_dynamic_library_t *library;
+  static iree_hal_mpi_dynamic_symbols_t symbols;
+  int rank;
+  int world_size;
+};
+
+iree_dynamic_library_t *LibmpiTest::library = NULL;
```

Comment:
style: asterisks on the left
```suggestion
iree_dynamic_library_t* LibmpiTest::library = NULL;
```

(if you add too many on the wrong side clang-format will move them all, so you may have put in one and then it "helped" by switching the rest :)

Reply:
hmm, really? :(